exe/CMakeLists.txt

```diff
@@ -9,6 +9,9 @@ add_executable(pxp-agent main.cc)
 target_link_libraries(pxp-agent libpxp-agent)
 install(TARGETS pxp-agent DESTINATION bin)
 
+install(FILES apply_ruby_shim.rb DESTINATION bin
+    PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
```

Comment:
I don't know that it's actually any sort of risk, but is there a reason to allow world read/execute?

Reply:
I copied those permissions from the ps1 helpers https://github.com/puppetlabs/pxp-agent/blob/main/exe/CMakeLists.txt#L34-L35 They also match the permissions for everything else in that directory

```
# ls -l /opt/puppetlabs/puppet/bin/
total 3588
-rwxr-xr-x 1 root root    4820 May  4 19:33 apply_ruby_shim.rb
-rwxr-xr-x 1 root root   19944 May  4 19:45 augmatch
-rwxr-xr-x 1 root root   11384 May  4 19:45 augparse
-rwxr-xr-x 1 root root   24200 May  4 19:45 augtool
-rwxr-xr-x 1 root root     544 Apr 28 18:12 bundle
-rwxr-xr-x 1 root root     546 Apr 28 18:12 bundler
-rwxr-xr-x 1 root root    6224 Apr 28 18:07 c_rehash
-rwxr-xr-x 1 root root  223200 May  4 19:44 curl
-rwxr-xr-x 1 root root    6071 Apr 28 18:10 curl-config
-rwxr-xr-x 1 root root    5078 Apr 28 18:12 erb
-rwxr-xr-x 1 root root  172192 May  4 19:44 execution_wrapper
lrwxrwxrwx 1 root root      53 May  4 20:21 eyaml -> /opt/puppetlabs/puppet/lib/ruby/vendor_gems/bin/eyaml
-rwxr-xr-x 1 root root     270 May  4 19:44 facter
-rwxr-xr-x 1 root root   15624 May  4 19:45 fadot
-rwxr-xr-x 1 root root     560 Apr 28 18:12 gem
-rwxr-xr-x 1 root root    6448 May  4 19:44 hiera
-rwxr-xr-x 1 root root     522 Apr 28 18:12 irb
-rwxr-xr-x 1 root root  653600 May  4 19:44 openssl
-rwxr-xr-x 1 root root     179 May  4 19:44 puppet
-rwxr-xr-x 1 root root     590 Apr 12 19:42 puppet-backup
-rwxr-xr-x 1 root root 2290880 May  4 19:44 pxp-agent
-rwxr-xr-x 1 root root     528 Apr 27 02:24 r10k
-rwxr-xr-x 1 root root     601 Apr 28 18:12 racc
-rwxr-xr-x 1 root root     605 Apr 28 18:12 racc2y
-rwxr-xr-x 1 root root     528 Apr 28 18:12 rake
-rwxr-xr-x 1 root root     528 Apr 28 18:12 rdoc
-rwxr-xr-x 1 root root     524 Apr 28 18:12 ri
-rwxr-xr-x 1 root root    7200 May  4 19:45 ruby
-rwxr-xr-x 1 root root     528 Apr 28 18:14 thor
-rwxr-xr-x 1 root root     939 May  4 19:44 wrapper.sh
-rwxr-xr-x 1 root root    1666 Apr 28 18:12 xml2-config
-rwxr-xr-x 1 root root   19808 May  4 19:44 xmlcatalog
-rwxr-xr-x 1 root root   75856 May  4 19:44 xmllint
-rwxr-xr-x 1 root root    2450 Apr 28 18:13 xslt-config
-rwxr-xr-x 1 root root   28200 May  4 19:44 xsltproc
-rwxr-xr-x 1 root root     605 Apr 28 18:12 y2racc
```

I don't _really_ know why, but it seems to be the norm.

Reply:
I guess not having that permission would create friction if we wanted to run pxp-agent as non-root. We don't support that now, but it seems like a reasonable practice for the installed binaries to be runnable by anyone.

Reply:
👍 cool, that's a reasonable reason